spec/lib/stripe/credit_card_cloner_spec.rb

```diff
@@ -5,62 +5,62 @@
+    describe "#find_or_clone", :vcr, :stripe_version do
+      include StripeStubs
 
-      let(:payment_method_response_body) {
-        JSON.generate(id: new_payment_method_id)
-      }
+      before { Stripe.api_key = secret }
 
-      before do
-        Stripe.api_key = "sk_test_12345"
+      let!(:customer_id) { ENV.fetch('STRIPE_CUSTOMER', nil) }
 
-        stub_customers_post_request email: credit_card.user.email,
-                                    response: { customer_id: new_customer_id },
-                                    stripe_account_header: true
+      let!(:stripe_account_id) { ENV.fetch('STRIPE_ACCOUNT', nil) }
 
-        stub_retrieve_payment_method_request(payment_method_id)
-        stub_list_customers_request(email: credit_card.user.email, response: {})
-        stub_get_customer_payment_methods_request(customer: "cus_A456", response: {})
-        stub_add_metadata_request(payment_method: "pm_456", response: {})
+      let!(:stripe_account) {
+        create(:stripe_account, enterprise:, stripe_user_id: stripe_account_id)
+      }
 
-        stub_request(:post,
-                     "https://api.stripe.com/v1/payment_methods/#{new_payment_method_id}/attach")
-          .with(body: { customer: new_customer_id },
-                headers: { 'Stripe-Account' => stripe_account_id })
-          .to_return(payment_method_response_mock)
+      let(:credit_card) { create(:credit_card, gateway_payment_profile_id: pm_card.id, user:) }
+
+      let(:pm_card) {
+        Stripe::PaymentMethod.create(
+          {
+            type: 'card',
+            card: {
+              number: '4242424242424242',
+              exp_month: 8,
+              exp_year: 2026,
+              cvc: '314',
+            },
+          },
+        )
+      }
 
-        credit_card.update_attribute :gateway_payment_profile_id, payment_method_id
+      let!(:connected_account) do
+        Stripe::Account.create({
+                                 type: 'standard',
+                                 country: 'AU',
+                                 email: 'jumping.jack@example.com'
+                               })
       end
 
+      let!(:cloner) { Stripe::CreditCardCloner.new(credit_card, connected_account.id) }
+
       context "when called with a card without a customer (one time usage card)" do
-        before do
-          stub_request(:post, "https://api.stripe.com/v1/payment_methods")
-            .with(body: { payment_method: payment_method_id },
-                  headers: { 'Stripe-Account' => stripe_account_id })
-            .to_return(payment_method_response_mock)
-        end
+        let!(:payment_method_id) { pm_card.id }
 
         it "clones the payment method only" do
-          customer_id, payment_method_id = cloner.find_or_clone
+          customer_id, new_payment_method_id = cloner.find_or_clone
 
-          expect(payment_method_id).to eq new_payment_method_id
+          expect(payment_method_id).not_to eq new_payment_method_id
```

Comment:
I wondering why the assertion changed here ? was the test before wrong ?

Reply:
I don't think it was wrong, but when using the stubs `new_payment_method_id` can be defined in advance and `payment_method_id` is changed by `cloner.find_or_clone`.

With the real HTTP calls, I found it more appropriate not to change `payment_method_id` and compare with `new_payment_method_id`, which is created when cloning, in this case.